pkg/broker/server/sse.go

```diff
@@ -244,6 +245,15 @@ func (b *Broker) run() {
 
 						panic("assert-failed")
 					}
+					if key != "/" {
+						// Make sure that the topic subscribed to by the client is the upper topic of the topic being notified or the topic exactly matched.
+						notfyTopic := strings.Split(event.topic, "/")
+						for i, sliceTopic := range strings.Split(key, "/") {
+							if notfyTopic[i] != sliceTopic {
+								return false
+							}
+						}
+					}
```

Comment:
Clever to do the check here.

Would you mind put this code in a function?   Something like  `if !checkPath(key, event.topic) { return false }`?